pkg/trace/stats/concentrator.go

```diff
@@ -113,20 +116,11 @@ type Input struct {
 }
 
 // NewStatsInput allocates a stats input for an incoming trace payload
-func NewStatsInput(numChunks int, containerID string, clientComputedStats bool, conf *config.AgentConfig) Input {
+func NewStatsInput(numChunks int, containerID string, clientComputedStats bool) Input {
 	if clientComputedStats {
 		return Input{}
 	}
-	in := Input{Traces: make([]traceutil.ProcessedTrace, 0, numChunks)}
-	_, enabledCIDStats := conf.Features["enable_cid_stats"]
-	_, disabledCIDStats := conf.Features["disable_cid_stats"]
-	enableContainers := enabledCIDStats || (conf.FargateOrchestrator != config.OrchestratorUnknown)
-	if enableContainers && !disabledCIDStats {
-		// only allow the ContainerID stats dimension if we're in a Fargate instance or it's
-		// been explicitly enabled and it's not prohibited by the disable_cid_stats feature flag.
-		in.ContainerID = containerID
-	}
-	return in
+	return Input{Traces: make([]traceutil.ProcessedTrace, 0, numChunks), ContainerID: containerID}
```

Comment:
why don't we check here the `conf.Features["disable_cid_stats"]` anymore (and/or Fargate presence), similar to what we do on the concentrator constructor?

Reply:
it's moved to the addStats in the concentrator, it was only checked there